File: app/src/main/java/com/skydroid/fpvtest/MainActivity.java
package com.skydroid.fpvtest;

import android.Manifest;
import android.content.Context;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;

public class  MainActivity extends AppCompatActivity {

    private Context context;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        this.context = this;
        requestPermissions();
        initView();
    }

    private void requestPermissions() {
        ActivityCompat.requestPermissions(
                this, new String[]{
                        Manifest.permission.READ_EXTERNAL_STORAGE,
                        Manifest.permission.WRITE_EXTERNAL_STORAGE
                }, 0
        );
    }

    private void initView(){
        findViewById(R.id.btn_h12_video).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                H12VideoSamplesActivity.start(context);
            }
        });

        findViewById(R.id.btn_usb_serial).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                UsbSerialActivity.start(context);
            }
        });
        findViewById(R.id.btn_h12_data).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                H12DataSamplesActivity.start(context);
            }
        });
    }
}

File: app/src/main/java/com/skydroid/fpvtest/MainActivity.kt
package com.skydroid.fpvtest

import android.Manifest
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.ActivityCompat

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        requestPermissions()
        initView()
    }

    private fun requestPermissions() {
        ActivityCompat.requestPermissions(
            this, arrayOf(
                Manifest.permission.READ_EXTERNAL_STORAGE,
                Manifest.permission.WRITE_EXTERNAL_STORAGE
            ), 0
        )
    }

    private fun initView() {
        findViewById<View>(R.id.btn_h12_video).setOnClickListener {
            H12VideoSamplesActivity.start(this@MainActivity)
        }
        findViewById<View>(R.id.btn_usb_serial).setOnClickListener {
            UsbSerialActivity.start(this@MainActivity)
        }
        findViewById<View>(R.id.btn_h12_data).setOnClickListener {
            H12DataSamplesActivity.start(this@MainActivity)
        }
        findViewById<View>(R.id.btn_h30_data).setOnClickListener {
            H30DataSamplesActivity.start(this@MainActivity)
        }
        findViewById<Button>(R.id.btn_frequency).setOnClickListener {
            startActivity(Intent(this@MainActivity, FrequencyActivity::class.java))
        }
    }
}